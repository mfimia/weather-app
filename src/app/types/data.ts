export interface IWeather {
  date: string
  temperature: number
  condition: string
  iconUrl: string
  humidity?: number
  windSpeed?: number
}

export interface UIComponent {
  type: string
  text: string
}

export interface CityWeatherData {
  [cityName: string]: IWeather[]
}

export interface UIData {
  ui: UIComponent[]
}

export type IWeatherData = CityWeatherData & UIData

export interface IWeatherPayload {
  cityName: Cities
  cityData: IWeather[]
  ui: UIComponent[]
}

export enum Cities {
  Düsseldorf = 'Düsseldorf',
  Málaga = 'Málaga',
  Berlin = 'Berlin'
}
