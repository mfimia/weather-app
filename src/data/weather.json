{
  "Düsseldorf": [
    {
      "date": "current",
      "temperature": 15,
      "condition": "Cloudy",
      "humidity": 68,
      "windSpeed": 10,
      "iconUrl": "/cloudy.svg"
    },
    {
      "date": "Monday",
      "temperature": 14,
      "condition": "Cloudy",
      "iconUrl": "/cloudy.svg"
    },
    {
      "date": "Tuesday",
      "temperature": 15,
      "condition": "Partly Cloudy",
      "iconUrl": "/partly_cloudy.svg"
    },
    {
      "date": "Wednesday",
      "temperature": 16,
      "condition": "Sunny",
      "iconUrl": "/sunny.svg"
    }
  ],
  "Málaga": [
    {
      "date": "current",
      "temperature": 25,
      "condition": "Sunny",
      "humidity": 50,
      "windSpeed": 5,
      "iconUrl": "/sunny.svg"
    },
    {
      "date": "Monday",
      "temperature": 26,
      "condition": "Partly Cloudy",
      "iconUrl": "/partly_cloudy.svg"
    },
    {
      "date": "Tuesday",
      "temperature": 25,
      "condition": "Sunny",
      "iconUrl": "/sunny.svg"
    },
    {
      "date": "Wednesday",
      "temperature": 24,
      "condition": "Cloudy",
      "iconUrl": "/cloudy.svg"
    }
  ],
  "Berlin": [
    {
      "date": "current",
      "temperature": 13,
      "condition": "Rainy",
      "humidity": 75,
      "windSpeed": 12,
      "iconUrl": "/rainy.svg"
    },
    {
      "date": "Monday",
      "temperature": 12,
      "condition": "Rainy",
      "iconUrl": "/rainy.svg"
    },
    {
      "date": "Tuesday",
      "temperature": 13,
      "condition": "Cloudy",
      "iconUrl": "/cloudy.svg"
    },
    {
      "date": "Wednesday",
      "temperature": 14,
      "condition": "Partly Cloudy",
      "iconUrl": "/partly_cloudy.svg"
    }
  ],
  "ui": [
    {
      "type": "header",
      "text": "Weather for {cityName}"
    },
    {
      "type": "weatherInfo"
    },
    {
      "type": "daysForecast"
    }
  ]
}