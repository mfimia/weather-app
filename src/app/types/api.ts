import { IWeatherPayload } from './data'

export interface IApiResponse {
  code: number
  data: IWeatherPayload
}
