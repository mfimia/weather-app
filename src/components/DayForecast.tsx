import { IWeather } from '@/app/types/data'
import Image from 'next/image'
import { FC } from 'react'

interface IDayForecastProps {
  forecast: IWeather
}

const DayForecast: FC<IDayForecastProps> = ({ forecast }) => {
  return (
    <div data-testid="dayForecast" className="p-2 border rounded shadow-sm">
      <Image
        className="mb-2"
        src={forecast.iconUrl}
        alt={forecast.condition}
        width={40}
        height={40}
      />
      <p className="text-gray-700 font-medium">{forecast.date}</p>
      <p>{`Temperature: ${forecast.temperature}Â°C`}</p>
      <p className="text-sm text-gray-600">{`Condition: ${forecast.condition}`}</p>
    </div>
  )
}

export default DayForecast
