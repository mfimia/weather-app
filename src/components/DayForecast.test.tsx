import { screen, render, waitFor } from '@testing-library/react'
import DayForecast from './DayForecast'
import { IWeather } from '@/app/types/data'

const mockForecast: IWeather = {
  date: '2023-10-06',
  temperature: 22,
  condition: 'Sunny',
  iconUrl: 'https://example.com/sunny-icon.png',
  humidity: 58,
  windSpeed: 10
}

describe('DayForecast', () => {
  it('should render correctly', () => {
    render(<DayForecast forecast={mockForecast} />)
    const dayForecastElement = screen.getByTestId('dayForecast')

    expect(dayForecastElement).toBeVisible()
    expect(dayForecastElement).toMatchSnapshot()
  })
})
