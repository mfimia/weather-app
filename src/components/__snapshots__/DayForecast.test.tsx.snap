// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayForecast should render correctly 1`] = `
<div
  class="p-2 border rounded shadow-sm"
  data-testid="dayForecast"
>
  <img
    alt="Sunny"
    class="mb-2"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/_next/image?url=https%3A%2F%2Fexample.com%2Fsunny-icon.png&w=96&q=75"
    srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fsunny-icon.png&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fsunny-icon.png&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
  <p
    class="text-gray-700 font-medium"
  >
    2023-10-06
  </p>
  <p>
    Temperature: 22Â°C
  </p>
  <p
    class="text-sm text-gray-600"
  >
    Condition: Sunny
  </p>
</div>
`;
